app/decorators/audio_work_show_decorator.rb

```diff
@@ -0,0 +1,36 @@
+class AudioWorkShowDecorator < WorkShowDecorator
+
+  # Used in works controller to figure out whether
+  # it's appropriate to show an audio playlist
+  # for a particular work.
+  # Doesn't care about the order of the members,
+  # just whether any of the published ones have playable audio.
+  def self.show_playlist?(some_work)
+    some_work.members.
+      where(published: true).
+      any? { | x| self.has_playable_audio_derivatives?(x) }
+  end
```

Comment:
No, this is a class method. I'm running it from the controller on *all* works (audio or not) to determine which decorator to use. It looks at all the derivatives and stops as soon as it finds *one* that's audio. `audio_members` isn't available in this context -- `audio_members` is an instance method to list *all* the audio members for an item that we've already determined that it has at least one playable audio track.

Reply:
Will update the comment! :)

Reply:
hmm, I guess I'm not sure why it's a class method! Maybe there's a larger refactor to be done, not sure. Bigger discussion maybe.

Reply:
It is doing an extra SQL fetch that I think doesn't have to and is best avoided, may require some refactoring to avoid, not sure, need to look at the code holistically and think about it.